typings/objection/index.d.ts

```diff
@@ -563,9 +564,7 @@ declare namespace Objection {
       ids: MaybeCompositeId | Partial<RelatedModel> | Partial<RelatedModel>[]
     ): NumberQueryBuilder<QB>;
 
-    <RelatedModel extends Model>(
-      ids: MaybeCompositeId | Partial<RelatedModel> | Partial<RelatedModel>[]
-    ): NumberQueryBuilder<QB>;
+    <RelatedModel extends Model>(ids: { [key: string]: MaybeCompositeId }): NumberQueryBuilder<QB>;
```

Comment:
I'm not sure how I feel about this. Do you have any thoughts? 

I added it to support the following case:

```ts
  await Person.relatedQuery('movies')
    .for(123)
    .relate({ foo: 50, bar: 20, baz: 10 });
```

Reply:
`Partial<RelatedModel>` already covers that case. The problem here is that `Person` doesn't have fields `foo`, `bar` or `baz`

Reply:
Makes sense. So those are intended to be id columns? I’d like to improve
the naming a bit since I don’t think it’s really clear

Reply:
This is currently the error I am seeing:

<img width="558" alt="Screen Shot 2019-09-27 at 8 56 58 AM" src="https://user-images.githubusercontent.com/2791730/65770786-d1f31b80-e104-11e9-8f59-f331ab7bae19.png">

It seems like it might not be inferring the model correctly since it is still typed as `Partial<Model>`

Reply:
Oh, that's true! There's no way for that to infer the model type. That should be something like

```ts
  interface RelateMethod<QB extends AnyQueryBuilder> {
    (ids: MaybeCompositeId | Partial<ModelType<QB>> | Partial<ModelType<QB>>[]): NumberQueryBuilder<
      QB
    >;
  }
```